backend/src/main/java/pt/ulisboa/tecnico/socialsoftware/tutor/user/UserService.java

```diff
@@ -153,4 +168,149 @@ public void resetDemoStudents() {
                     this.userRepository.delete(user);
                 });
     }
+
+    @Retryable(
+            value = { SQLException.class },
+            backoff = @Backoff(delay = 5000))
+    @Transactional(isolation = Isolation.READ_COMMITTED)
+    public void importListOfUsers(InputStream stream, int courseExecutionId) {
+        String line = "";
+        String cvsSplitBy = ",";
+        User.Role auxRole;
+        InputStreamReader isr = new InputStreamReader(stream, StandardCharsets.UTF_8);
+        try (BufferedReader br = new BufferedReader(isr)) {
+            while ((line = br.readLine()) != null) {
+                String[] userInfo = line.split(cvsSplitBy);
+                if (userInfo.length == 2) {
+                    auxRole = User.Role.STUDENT;
+                }
+                else if (userInfo.length > 2) {
```

Comment:
auxRole = User.Role.valueOf(userInfo[2]); and check possible errors when string value is incorrect